docs/sdks/java.md

```diff
@@ -31,136 +34,214 @@ There's an optional uber-jar with shaded and relocated guava and failsafe depend
+    return "Feature flag value of %s is %s".formatted(key, val);
+  }
+}
 ```
 
-## Get Config
+Feature flags don't have to return just true or false. You can get other data types using `get`:
 
 ```java
-final Optional<Prefab.ConfigValue> configValue = prefabCloudClient.configClient().get("the.key");
-if(configValue.isPresent()){
-  System.out.println(configValue.get().getString());
+public class MyClass {
+  // assumes you have setup a singleton
+  @Inject
+  private FeatureFlagClient featureFlagClient;
+
+  public String test(String key){
+    // highlight-next-line
+    Optional<Prefab.ConfigValue> val = featureFlagClient.get(key);
+    return "Feature flag value of %s is %s".formatted(key, val.orElse("no value found"));
+  }
 }
 ```
 
-## Provide Context
+## Context
 
-To finely-target configuration rule evaluation, we accept contextual information both inline when making a get request
+To finely-target configuration rule evaluation, we accept contextual information globally or request-scoped with the ContextStore which will affect all logging, featureflag and config lookups.
 
 ```java
-prefabCloudClient.configClient().get("the.key", PrefabContext.newBuilder("user")
-                                                    .put("name", "james")
-                                                    .put("tier", "gold")
-                                                    .put("customerMonths", 12)
-                                                    .build()
-                                                 )
+prefabCloudClient.configClient().getContextStore().addContext(PrefabContext.newBuilder("User")
+                        .put("name", user.getName())
+                        .build());
 ```
 
-and globally or request-scoped with the ContextStore which will affect all logging, featureflag and config lookups.
+When global context is set, log levels and feature flags will evaluate in that context. Here are details on setting global context:
+
+<Tabs groupId="lang">
+<TabItem value="micronaut" label="Micronaut">
+
+In your Prefab options, set your Prefab Context store to be your ServerRequestContextStore.
 
 ```java
+options.setContextStore(new ServerRequestContextStore());
+```
 
-prefabCloudClient.configClient().getContextStore().addContext(PrefabContext.newBuilder("User")
+Next we add a [filter](https://github.com/prefab-cloud/example-micronaut-app/blob/configure-prefab-context/src/main/java/com/example/prefab/PrefabContextFilter.java) to add a prefab context based on the currently "logged in" user.
+
+```java
+configClient.getContextStore()
+      .addContext(PrefabContext.newBuilder("user")
+          .put("id", user.id())
+          .put("country", user.country())
+          .put("email", user.email())
+          .build()
+      );
+```
+
+Learn more with the [Prefab + Micronaut example app](https://github.com/prefab-cloud/example-micronaut-app)
+
+</TabItem>
+
+<TabItem value="dropwizard" label="Dropwizard">
```

Comment:
I don't think we need the drop wizard vs micronaut things here - the snippet above isn't fully developed (we could copy paste more) compared to the one below , which may be over developed and too long, plus we can link to the full versions